jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2388,17 +2406,31 @@ public boolean equalsIgnoreCase(String string) {
 				// overloads rather than code points.
 				/*[ENDIF]*/
 				if (byteAtO1 != byteAtO2 
-						&& toUpperCase(helpers.byteToCharUnsigned(byteAtO1)) != toUpperCase(helpers.byteToCharUnsigned(byteAtO2))
-						&& toLowerCase(helpers.byteToCharUnsigned(byteAtO1)) != toLowerCase(helpers.byteToCharUnsigned(byteAtO2))) {
+						&& toUpperCase(helpers.byteToCharUnsigned(byteAtO1)) != toUpperCase(helpers.byteToCharUnsigned(byteAtO2))) {
 					return false;
 				}
 			}
 		} else {
-			while (o1 < end) {
+			// Compare the last chars.
+			// In order to tell 2 chars are different:
+			// If at least one char is ASCII, converting to upper cases then compare should be sufficient.
+			// If both chars are not in ASCII char set, need to convert to lower case and compare as well.
+			char charAtO1Last = s1.charAtInternal(s1len-1, s1Value);
+			char charAtO2Last = s2.charAtInternal(s1len-1, s2Value);
+
+			if (charAtO1Last != charAtO2Last 
+					&& toUpperCase(charAtO1Last) != toUpperCase(charAtO2Last)
+					&& (charAtO1Last < 128 || charAtO2Last < 128 || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last))) {
```

Comment:
Since compressed characters (the first 256) don't need a lower case check, this check can at least use 256.

Reply:
According to my testing, 304 would also work.

Reply:
I'm fine with extending the check to 304 (and possibly further). But if the chars between 256 and 304 are not often used, I prefer to leave at 256 since it's more clear and 304 seems a little random.

Reply:
I guess it depends what language you are using and what characters you need. I won't push for 304. I wanted to double check the first 256 and I extended my test to see how far it could go before finding a problem, 304 was the first character that needed the toLowerCase() check.